### ISSUE TITLE
prepare next release

### ISSUE THREAD

**Issue by ijgnd** [ankipalace/advanced-browser#156]
- the columns "Card (internal)->Card Queue" and "Card (internal)->Card Type" were broken in the versions that I tested with (Anki 2.1.45 and 2.1.54). I fixed this in the master branch.
- I fixed a deprecation warning in the master branch.

Additionally I added two new columns:

- Total Again Count: the built-in Lapses column only counts again for review cards. I wanted a column that counts every Again-rating (should be useful e.g. when I have many long learning steps, see #155). This in the branch "again_count"
- "% of scheduled interval": for my motivation see #154. This in the branch "percent_of_scheduled_ivl". My code for calculating the relative dueness probably doesn't take into account all possible combinations of card queues and card types. But so far it seems to work well enough for me.

@AnKingMed: After your holiday have a look at my two new columns, do a quick test on 2.1.45+ (you need to checkout these branches separately and test them individually) and see if you want them included into the AB browser add-on. Then I can merge these branches and build a new release for 2.1.45 that you can upload to ankiweb.

**Comment by ijgnd** [comment]
to make testing more convenient: here is one file that contains the changes from both branches mentioned in the first post. This is for .45+ only. Since github does not allow to upload files with the extension ".ankiweb" I added ".zip" to the filename which you need to remove before you install it into Anki.

When testing keep in mind that "% of scheduled interval" only works for cards in review that are not in relearing (search term: is:review -is:learn"). For me it doesn't make sense to calculate the overdueness of learn cards by the minute. i also calculate no value for cards that you relearn. That was easier to code and the question is which is the relevant overdue interval: in reference to your relearning steps or the interval that you get after this with the "new interval" setting?

About the Total again count: if it's zero the column is empty.

[Advanced_Browser__branch_idx_4__AnkiVers_45-54__2022-10-08__both_new_features_merged_ijgnd.ankiaddon.zip](https://github.com/ankipalace/advanced-browser/files/9738749/Advanced_Browser__branch_idx_4__AnkiVers_45-54__2022-10-08__both_new_features_merged_ijgnd.ankiaddon.zip)

**Comment by ijgnd** [comment]
@AnKingMed : see https://github.com/ankipalace/advanced-browser/issues/156#issuecomment-1272263942

**Comment by ijgnd** [comment]
@AnKingMed: when you have some time left over: In October I added two new columns to the Advanced Browser: "% of scheduled interval"  and "Total Again Count". If you like these the code needs some testing before this version can be released. the ankiaddon file is in the second post in this thread.

**Comment by AnKingMed** [comment]
These are really cool. I tested and things look great. Again count seems very useful to me. I'm not sure I understand "% of scheduled interval".  
Could you explain this more and how one would use it?

**Comment by ijgnd** [comment]
My motivation for  "% of scheduled interval":

this column might be useful if you want to study your most overdue cards in random order. You can sort filtered decks by relative overdueness. But if you created two notes with similar content and studied them only once or twice they will have seen little fuzz to their intervals so that they'll have very similar overdue values so that when sorting the cards by relative overdueness they'll be shown closely together. I prefer fully randomized reviews that I can get by building a filtered deck from selected cards in the browser when the browser table is sorted by  "% of scheduled interval".

I originally had the code study the most "underdue" cards. But for this use case now there's the add-on [Smarter Study Ahead](https://ankiweb.net/shared/info/1536096830)

&nbsp;

I think you could release them now. But releasing them is not important since very few people will use these, I guess. I mainly cared if they can be merged into the main branch of this add-on so that they are automatically included in the next release so that I don't have to build custom versions each time a new Advanced Browser version is released. I thought that 

I asked you about checking the commits now because I still remember making these commits ...

**Comment by AnKingMed** [comment]
What is the actual calculation you are using for % of scheduled interval? (Just curiosity)

Also I'm totally fine merging this in and releasing it now!

**Comment by ijgnd** [comment]
I use [this](https://github.com/ankipalace/advanced-browser/blob/6d3e2b3a7874b5c892a5bb10c7df779f5a4bc4c3/advancedbrowser/advancedbrowser/advanced_fields.py#L406):

    last_rev = card.due - card.ivl
    elapsed = mw.col.sched.today - last_rev
    val = elapsed/float(card.ivl) * 100
    ....
    fmt = "{0:.2f}".format(val) + " %"

**Comment by AnKingMed** [comment]
Awesome. Thanks! I was just curious.   You can go ahead and merge the branches. You should have permissions

**Comment by ijgnd** [comment]
Thanks for your review. I've merged these. This is not my add-on so I prefer getting a second opinion.

**Comment by AnKingMed** [comment]
Should I just upload the file you sent here to AnkiWeb? for 2.1.45+ correct?

**Comment by ijgnd** [comment]
it's 2.1.45+

to be on the safe side I'd like to check this on the weekend.

**Comment by AnKingMed** [comment]
Ok let me know. If you could just upload a new release I will download that file and release it

**Comment by ijgnd** [comment]
yes [the file I posted on 2022-10-08](https://github.com/ankipalace/advanced-browser/issues/156#issuecomment-1272263942) can be uploaded for 2.1.45.

It's identical to building from the master branch of this addon with `python3 release.py` and it's what I've been using the last two months.

**Comment by ijgnd** [comment]
off-topic: the download link for the addon newly released "Spell Checker"  on ankipalace.com is broken. It leads to an error page and if you extract the bitly part it leads to the weekends add-on

**Comment by AnKingMed** [comment]
Just fixed this. Can you try again?

**Comment by ijgnd** [comment]
the spell checker download link now works as expected

**Comment by AnKingMed** [comment]
Great. Thanks for reporting that!
